Batteries/Data/Char.lean

```diff
@@ -11,3 +12,73 @@ theorem Char.le_antisymm_iff {x y : Char} : x = y ↔ x ≤ y ∧ y ≤ x :=
 
 instance : Batteries.LawfulOrd Char := .compareOfLessAndEq
   (fun _ => Nat.lt_irrefl _) Nat.lt_trans Nat.not_lt Char.le_antisymm
+
+namespace Char
+
+/--
+Unicode defines a "case folding" algorithm in which upper-case ASCII characters are
+mapped onto their lower-case counterparts.
+
+This abbrev standardizes on one direction (it answers the question of whether
+one should choose e.g. `toUpper` or `toLower` with respect to case insensitive logic).
+-/
+abbrev asciiCaseFold := Char.toLower
```

Comment:
```suggestion
/-- Case folding for ASCII characters only.

Alphabetic ASCII characters are mapped to their lowercase form, all other characters are left unchanged. This agrees with the Unicode case folding algorithm for ASCII characters.

```
#eval caseFoldAsciiOnly 'A' == 'a'
#eval caseFoldAsciiOnly 'a' == 'a'
#eval caseFoldAsciiOnly 'À` == 'À`
#eval caseFoldAsciiOnly `à` == `à`
#eval caseFoldAsciiOnly `$` == `$`
```
-/
abbrev caseFoldAsciiOnly := Char.toLower
```

Reply:
I think the easiest naming scheme for this PR is "basic name" + `AsciiOnly`, where "basic name" is the name that would make sense if it was a full Unicode implementation (or if there were only ASCII characters). The docstring should explain what `AsciiOnly` precisely means, with examples.

Reply:
That works for me, thanks. I've made the appropriate changes.